provisioning/device/test/_transport_state_machine_test.js

```diff
@@ -0,0 +1,502 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var TransportStateMachine = require('../lib/transport_state_machine').TransportStateMachine;
+var sinon = require('sinon');
+var assert = require('chai').assert;
+
+var fakeErrorText = '__FAKE_ERROR__';
+var fakeRegistrationId = '__FAKE_REGISTRATION_ID__';
+var fakeAuth = '__FAKE_AUTH__';
+var fakeRequestBody = { fake: true };
+var fakeBadStatus = '__FAKE_BAD_STATUS__';
+
+var waitingForNetworkIo = function () { return sinon.spy(function () { }); };
+var registrationRequestReturnsAssigning = function (pollingInterval) { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(null, { status: 'Assigning' }, null, pollingInterval || 0); }); };
+var registrationRequestReturnsAssigned = function () { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(null, { status: 'Assigned' }); }); };
+var registrationRequestReturnsFailure = function () { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(new Error(fakeErrorText)); }); };
+var registrationRequestReturnsBadResponse = function () { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(null, { status: fakeBadStatus }); }); };
+var operationStatusReturnsAssigned = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(null, { status: 'Assigned' } ); }); };
+var operationStatusReturnsFailure = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(new Error(fakeErrorText)); }); };
+var operationStatusReturnsBadResponse = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(null, { status: fakeBadStatus }); }); };
+var operationStatusReturnsAssigningThenAssigned = function () {
+  var callCount = 0;
+  return sinon.spy(function (registrationId, operationId, callback) {
+    callCount++;
+    if (callCount === 1) {
+      callback(null, { status: 'Assigning' }, null, 0);
+    } else if (callCount === 2) {
+      callback(null, { status: 'Assigned' });
+    } else {
+      assert(false, 'second status query returned Assigned.  There should not be a third status query');
+    }
+  });
+};
+
+describe('state machine', function () {
+  this.timeout(1000);
+
+  var makeNewMachine = function () {
```

Comment:
>makeNewMachine [](start = 6, length = 14)

why not use beforeEach() and afterEach() hooks to reinitialize everything between tests?